solutions/platinum/usaco-1332.mdx

```diff
@@ -0,0 +1,265 @@
+---
+id: usaco-1332
+source: USACO Platinum 2023 March
+title: Pareidolia
+author: Aakash Gokhale
+---
+
+## Explanation
+
+Let's first try to solve the problem without updates on a single string.
+Notice that it is always optimal to take the earliest $b$ and then the earliest $e$ and then the earliest $s$ after that and so on because it maximizes the number of places you can search to find the next letter of "bessie".
+
+Now let's try to solve the problem without updates for all substrings.
+Let $dp[i]$ be the number of positions of the string we have processed so far such that if we apply the greedy strategy described above, the next character we need to add is the $i$th character ($0$ indexed) of the string "bessie". We add $1$ to $dp[0]$ for every position of the string which represents the substring starting from the current index.
```

Comment:
```suggestion
First, let's solve the problem without updating a single string.

Notice that it is always optimal to take the earliest $b$, then the earliest $e$, then the earliest $s$ after that and so on because it maximizes the number of places you can search to find the following letter of "bessie."
		
Now try solving the problem without updates for all substrings.

Let $dp[i]$ be the number of positions of the string processed so far such that if we apply the greedy strategy described above, the next character we need to add is the $i$th character ($0$ indexed) of the string "bessie."

We add $1$ to $dp[0]$ for every string position, representing the substring starting from the current index.
```

Reply:
This sentence "We add $1$ to $dp[0]$ for every string position, representing the substring starting from the current index." doesn't really make sense

Reply:
Is it better now?

Reply:
Yes